eng/fv/methodology.rst

```diff
@@ -0,0 +1,53 @@
+.. SPDX-License-Identifier: CC-BY-SA-4.0
+
+.. Copyright (C) 2022 Trinity College Dublin
+
+Test Generation Methodology
+===========================
+
+The general approach to using any model-checking technology for test generation
+has three major steps:
+
+Model desired behavior
+----------------------
+
+Construct a model that describes the desired properties (`P1`, ..., `PN`)
+and use the model-checker to verify those properties.
+
+Promela can specify properties using the ``assert()`` statement, to be
+true at the point where it gets executed, and can use Linear Temporal Logic
+(LTL) to specify more complex properties over execution sequences. SPIN will
+also check generic correctness properties such as deadlock and
+livelock freedom.
+
+Make claims about undesired behavior
+------------------------------------
+
+Given a fully verified model, systematically negate each specified property.
+Given that each property was verified as true, 
+then these negated properties will fail model-checking, 
+and counter-examples will be
+generated. These counter-examples will in fact be scenarios describing correct
+behavior of the system, demonstrating the truth of each property.
+
+.. warning::
+
+  It is very important that the negations only apply to stated properties,
+  and do not alter the possible behaviors of the model in any way.
```

Comment:
Indeed - I'll add material to elaborate on this

Reply:
Added some extra material: basically control-flow conditions should not be changed,
so that only allows assert statements to be negated, as well as LTL properties.